src/Compilers/CSharp/Test/Syntax/IncrementalParsing/IncrementalParsingTests.cs

```diff
@@ -584,6 +588,267 @@ int LocalFunc()
             WalkTreeAndVerify(tree.GetCompilationUnitRoot(), fullTree.GetCompilationUnitRoot());
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/74456")]
+        public void TestCollectionExpressionSpreadVsDeletingTopLevelBrace()
+        {
+            const string valueSetterLine = "x[1] = 312;";
+            var initialSource = $$"""
+                public class Program
+                {
+                    public void M2()
+                    {
+                        if (true)
+                        {
+                            {
+                                if (true)
+                                {
+                                    {{valueSetterLine}}
+                                }
+                            }
+                        }
+                        if (true)
+                        {
+                            y = [.. z];
+                        }
+                    }
+                }
+                """;
+            var initialTree = SyntaxFactory.ParseSyntaxTree(initialSource);
+
+            // Initial code is fully legal and should have no parse errors.
+            Assert.Empty(initialTree.GetDiagnostics());
+
+            // Delete '{' (and end of line) before 'values[1] = 312;'
+            var initialText = initialTree.GetText();
+            var valueSetterLinePosition = initialSource.IndexOf(valueSetterLine);
+            var initialLines = initialText.Lines;
+
+            int valueSetterLineIndex = initialLines.IndexOf(valueSetterLinePosition);
+            var openBraceLine = initialText.Lines[valueSetterLineIndex - 1];
+            Assert.EndsWith("{", openBraceLine.ToString());
+
+            var withOpenBraceDeletedText = initialText.WithChanges(new TextChange(openBraceLine.SpanIncludingLineBreak, ""));
+            var withOpenBraceDeletedTree = initialTree.WithChangedText(withOpenBraceDeletedText);
+
+            // Deletion of the open brace causes the method body to close early with the close brace before the `if`
+            // statement.  This will lead to a ton of cascading errors for what follows.  In particular, the `[.. values]`
+            // will be parsed as a very broken attribute on an incomplete member.
+            {
+                UsingTree(withOpenBraceDeletedTree,
```

Comment:
I like having this in the test so that if something changes int eh future we can validate that that's ok.  these incremental scenarios are complex, and understanding the shape of things when things break is invaluable.